sql/log_event.h

```diff
@@ -5283,6 +5283,61 @@ class Delete_rows_compressed_log_event : public Delete_rows_log_event
 #endif
 };
 
+/**
+  @class Wsrep_write_set_meta_log_event
+
+  Log the write set metadata for the WSREP plugin. The event is logged
+  by Galera node before XID event.
+
+  <table id="Wsrep_write_set_meta_log_event_format">
+    <caption>Wsrep_write_set_meta_log_event format</caption>
+    <tr>
+      <th>Symbol</th>
+      <th>Format</th>
+      <th>Description</th>
+    </tr>
+    <tr>
+      <td>SEQNO</td>
+      <td align="right">8</td>
+      <td>Sequence number of the write set</td>
+    </tr>
+    <tr>
+      <td>UUID</td>
+      <td align="right">16</td>
+      <td>UUID of the write set</td>
+    </tr>
+    <tr>
+      <td>FLAGS</td>
+      <td align="right">1</td>
+      <td>Flags field reserved for future use</td>
+    </tr>
+  </table>
+*/
+#define WSREP_WRITE_SET_META_EVENT_DATA_LEN 24
+class Wsrep_write_set_meta_log_event : public Log_event
+{
+public:
+  int64 seqno;
+  uchar uuid[16];
```

Comment:
The group uuid rarely changes, and storing it into each and every time adds 16-byte overhead per transaction. Consider splitting the meta data log event into `Wsrep_seqno_log_event` and `Wsrep_group_uuid_log_event`. The `Wsrep_group_id_log_event` need to be logged every time the group UUID changes or when binlog is rotated.